CONTRIBUTING.txt

```diff
@@ -507,6 +508,24 @@ cases, the developers can agree on a different procedure upon justification
 (like when we can't detect the change, or it involves moving or deleting an
 entire function for example).
 
+As warnings are rather intrusive, their message and placement is designed to
+guide users to take immediate action so that important changes
+will not break their code in a future release.
+The ``stacklevel`` argument in the ``warn``
+`function <https://docs.python.org/3/library/warnings.html#warnings.warn>`_
+should be used in order to ensure the warning point to the user's line of code,
+where they have the ability to take an action.
+In most cases, it is appropriate to set the ``stacklevel`` to ``2``, but cases
+when warnings originate from helper routines internal to the scikit-image
+library, it is may be more appropriate to set the ``stacklevel`` to ``3``.
+
+To test if your warning is being emitted correctly, try calling the function
+from an python console. You it should point you to the python console itself
+instead of being emitted by the files in the scikit-image library
+
+* **Good**: ``ipython:1: UserWarning: ...``
+* **Bad**: ``scikit-image/skimage/measure/_structural_similarity.py:155: UserWarning:``
```

Comment:
@stefanv is this a good enough description

Reply:
This is good, but nowhere is there a hint on what the stacklevel argument *means*.

The first sentence is also a bit long.  Perhaps, instead of:

> As warnings are rather intrusive, their message and placement is designed to
guide users to take immediate action so that important changes
will not break their code in a future release.

Just:

> scikit-image uses warnings to highlight changes in its API, so that users may update their code accordingly. The ``stacklevel`` ...

Reply:
hopefully that addresses things.